fix(ivy): injecting incorrect Renderer2 into child components

crisbeto opened the pull request:
In ViewEngine injecting a Renderer2 returns a renderer that is specific to the particular component, however in Ivy we inject the renderer for the parent view instead. This causes it to set incorrect `ngcontent` attributes when creating elements through the renderer.

The issue comes from the fact that the `Renderer2` is created according to the current `LView`, but because DI happens before we've entered the `LView` of the component that's injecting the renderer, we end up with one that's one level up. We work around the issue by finding the `LView` that corresponds to the `previousOrParentTNode` inside of the parent view and associating the `Renderer2` with it.

This PR resolves FW-1382.

kara commented on packages/core/src/render3/view_engine_compatibility.ts:
```diff
@@ -384,7 +385,23 @@ function getOrCreateRenderer2(view: LView): Renderer2 {
+  // We need the Renderer to be based on the component that it's being injected into, however since
+  // DI happens before we've entered its view, `getLView` will return the parent view instead.
+  // We work around it by walking through the parent view and looking for a child view that
+  // matches the TNode.
```
Since we have the node index, I think we should be able to do this instead of looping:

```ts
const componentView = getComponentViewByIndex(tNode.index);
```

crisbeto replied:
Ah nice, that's what I was looking for before settling for the loop.

jelbourn reviewed (approved):
LGTM

crisbeto commented:
Reworked based on the feedback @kara.

crisbeto committed:
fix(ivy): injecting incorrect Renderer2 into child components

In ViewEngine injecting a Renderer2 returns a renderer that is specific to the particular component, however in Ivy we inject the renderer for the parent view instead. This causes it to set incorrect `ngcontent` attributes when creating elements through the renderer.

The issue comes from the fact that the `Renderer2` is created according to the current `LView`, but because DI happens before we've entered the `LView` of the component that's injecting the renderer, we end up with one that's one level up. We work around the issue by finding the `LView` that corresponds to the `previousOrParentTNode` inside of the parent view and associating the `Renderer2` with it.

This PR resolves FW-1382.

kara reviewed (approved):
LGTM

kara commented:
[VE presubmit](https://test.corp.google.com/ui#id=OCL:254785402:BASE:254785447:1561397397599:aa6e9a44)

[Ivy presubmit](http://test/OCL:254785402:BASE:254788713:1561398208653:15fc98a8)
